src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ReferencesPage/ImportedNamespacesValueProvider.cs

```diff
@@ -40,64 +41,57 @@ private async Task<ImmutableArray<string>> GetSelectedImportsAsync()
     
     private async Task<string> GetSelectedImportStringAsync()
     {
-        StringBuilder sb = new StringBuilder();
         string projectName = Path.GetFileNameWithoutExtension(_configuredProject.UnconfiguredProject.FullPath);
         bool containsProjectName = false;
 
-        ImmutableArray<string> existingImports = await GetSelectedImportsAsync();
-        
-        foreach (string? import in existingImports)
-        {
-            if (!string.IsNullOrEmpty(import))
-            {
-                if (string.Equals(import, projectName))
-                {
-                    containsProjectName = true;
-                }
-                sb.Append(import).Append(';');
-            }
-        }
+        ImmutableArray<(string Value, bool IsImported)> existingImports = await GetSelectedImportsAsync();
+
+        var selectedImports = existingImports
+                .Where(pair => !string.IsNullOrEmpty(pair.Value))
+                .Select(pair => (Key: pair.Value, Value: pair.IsImported.ToString())).ToList();
 
         if (!containsProjectName)
         {
-            sb.Append(projectName).Append(';');
+            selectedImports.Add((projectName, Value: bool.TrueString));
         }
 
-        sb.Remove(sb.Length - 1, 1);
-        return sb.ToString();
+        return _encoding.Format(selectedImports);
     }
     
-    public override Task<string> OnGetUnevaluatedPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
+    public override async Task<string> OnGetUnevaluatedPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
     {
-        return GetSelectedImportStringAsync();
+        return await GetSelectedImportStringAsync();
     }
 
-    public override Task<string> OnGetEvaluatedPropertyValueAsync(string propertyName, string evaluatedPropertyValue, IProjectProperties defaultProperties)
+    public override async Task<string> OnGetEvaluatedPropertyValueAsync(string propertyName, string evaluatedPropertyValue, IProjectProperties defaultProperties)
     {
-        return GetSelectedImportStringAsync();
+        return await GetSelectedImportStringAsync();
```

Comment:
There are pros and cons to this change, but it does add an extra `Task<>` allocation for each call. The pro is that this method would appear in any async stacks, whereas if you just returned the task directly (rather than being async and awaiting it) you'd see this method in the continuation chain.

Reply:
What would you recommend?

Reply:
We generally favour avoiding the allocation, but it's not a huge deal in this case as it's not a tight loop. Feel free to leave it.